test/hotspot/jtreg/compiler/vectorapi/TestVectorShuffleIotaByte.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import jdk.incubator.vector.ByteVector;
+import jdk.incubator.vector.VectorSpecies;
+import jdk.incubator.vector.VectorShuffle;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+/*
+ * @test
+ * @bug 8265956
+ * @modules jdk.incubator.vector
+ * @run testng/othervm compiler.vectorapi.TestVectorShuffleIotaByte
+ */
+
+@Test
+public class TestVectorShuffleIotaByte {
+    static final VectorSpecies<Byte> SPECIESb_64 = ByteVector.SPECIES_64;
+    static final VectorSpecies<Byte> SPECIESb_128 = ByteVector.SPECIES_128;
+    static final VectorSpecies<Byte> SPECIESb_256 = ByteVector.SPECIES_256;
+    static final VectorSpecies<Byte> SPECIESb_512 = ByteVector.SPECIES_512;
+
+    static final int INVOC_COUNT = Integer.getInteger("jdk.incubator.vector.test.loop-iterations", 50000);
+
+    static final byte[] ab_64 = {41, 45, 59, 46, 115, 101, 103, 97};
+    static final byte[] ab_128 = {112, 32, 116, 117, 111, 104, 116, 105, 119, 32, 107, 111, 111, 98, 32, 97};
+    static final byte[] ab_256 = {32, 101, 107, 105, 108, 32, 115, 105, 32, 117, 111, 121, 32, 116, 117, 111,
+                                  104, 116, 105, 119, 32, 121, 97, 100, 32, 121, 114, 101, 118, 69, 32, 46};
+    static final byte[] ab_512 = {103, 110, 97, 117, 72, 32, 71, 78, 65, 87, 32, 45, 45, 33, 117, 111, 121, 32,
+                                  103, 110, 105, 115, 115, 105, 77, 32, 46, 117, 111, 121, 32, 111, 116, 32, 114,
+                                  101, 116, 116, 101, 108, 32, 100, 114, 105, 104, 116, 32, 121, 109, 32, 115, 105,
+                                  32, 115, 105, 104, 116, 44, 121, 116, 101, 101, 119, 83};
+
+    static final byte[] expected_64 = {1, 3, 5, 7, -7, -5, -3, -1};
+    static final byte[] expected_128 = {1, 3, 5, 7, 9, 11, 13, 15, -15, -13, -11, -9, -7, -5, -3, -1};
+    static final byte[] expected_256 = {1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31,
+                                        -31, -29, -27, -25, -23, -21, -19, -17, -15, -13, -11, -9, -7, -5, -3, -1};
+    static final byte[] expected_512 = {1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31,
+                                        33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63,
+                                        -63, -61, -59, -57, -55, -53, -51, -49, -47, -45, -43, -41, -39, -37, -35, -33,
+                                        -31, -29, -27, -25, -23, -21, -19, -17, -15, -13, -11, -9, -7, -5, -3, -1};
+
+    @Test
+    static void testShuffleIota_64() {
+        for (int ic = 0; ic < INVOC_COUNT; ic++) {
+            ByteVector bv1 = (ByteVector) VectorShuffle.iota(SPECIESb_64, 1, 2, false).toVector();
+            bv1.intoArray(ab_64, 0);
+        }
+        Assert.assertEquals(ab_64, expected_64);
```

Comment:
How about putting the expected value as the first parameter, which is suggested by [1] ?
 
[1] https://github.com/openjdk/jdk/blob/master/doc/hotspot-unit-tests.md#first-parameter-is-expected-value

Reply:
Thank you for your review.
* I think this document might be the advice of GoogleTest in jdk(`test/hotspot/gtest`) instead of jtreg (`test/hotspot/jtreg`)
* The common way which can pass expect result to method is using `@DataProvider`. However, in this small test it will make the whole test case inflate. :-)